docs/resources/grants.md

```diff
@@ -194,6 +194,24 @@ resource "databricks_grants" "volume" {
 }
 ```
 
+## Function grants
+
+You can grant `ALL_PRIVILEGES`, `APPLY_TAG`, and `EXECUTE` privileges to [_`catalog.schema.function`_](registered_model.md) specified in the `function` attribute.
+
+```hcl
+resource "databricks_grants" "customers" {
+  function = "main.reporting.customer_model"
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["APPLY_TAG", "EXECUTE"]
+  }
+  grant {
+    principal  = "Data Analysts"
+    privileges = ["EXECUTE"]
+  }
+}
+```
```

Comment:
Why it's called `function`? Won't be people confused by it?

Reply:
I think this is because the underlying securable_type (which is the field used to construct the [UC permissions REST API](https://docs.databricks.com/api/workspace/grants/update) path) is `function`, but I don't see any downside to calling the resource `model` here. Similarly in SQL we'll support things like `GRANT EXECUTE ON MODEL mycatalog.myschema.mymodel`.

Reply:
Unfortunately the underlying [docs](https://docs.databricks.com/api/workspace/grants/get)/Go SDK don't have `model` in the enum because as you said it's not a securable type, so we get the failure seen below.
<img width="662" alt="Screenshot 2023-10-06 at 10 55 05 AM" src="https://github.com/databricks/terraform-provider-databricks/assets/87999496/2a199bd3-e2b0-4104-8f84-a45a0f2ed3b0">

We could create a custom mapping from `model` to `function`, but that might be weird/complicate the code. I don't think this is that odd considering our [docs page](https://docs.databricks.com/api/workspace/registeredmodels) also has a note to use `function` for the securable type.

```
Note: The securable type for models is "FUNCTION". When using REST APIs (e.g. tagging, grants) that specify a securable type, use "FUNCTION" as the securable type.
```

Reply:
we already had to build custom mapping for a few securables, so it shouldn't be too much work - https://github.com/databricks/terraform-provider-databricks/blob/master/catalog/resource_grants.go#L101

Reply:
@nkvuong that's perfect! Added `model` there and it seems to work now